snapcraft/tests/test_commands_search.py

```diff
@@ -0,0 +1,73 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import io
+import logging
+from unittest import mock
+
+import fixtures
+
+from snapcraft import main, tests
+from snapcraft.internal import parts
+from snapcraft.tests import fixture_setup
+
+
+class SearchCommandTestCase(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.useFixture(fixture_setup.FakeParts())
+        with mock.patch('snapcraft.internal.parts.ProgressBar',
+                        new=tests.SilentProgressBar):
+            parts.update()
+
+        patcher = mock.patch('snapcraft.internal.parts.get_terminal_width')
+        self.mock_terminal = patcher.start()
+        self.mock_terminal.return_value = 80
+        self.addCleanup(patcher.stop)
+
+    @mock.patch('sys.stdout', new_callable=io.StringIO)
+    def test_searching_for_a_part_that_exists(self, mock_stdout):
```

Comment:
you are missing a test for a match in description, not in name.
And for completenes, maybe a match in name, not in description.